jazzband/django-silk
# Support Pre-Commit CI

Jazzband org has added pre-commit CI as a requirement.

Using the [file suggested by Andrew Chen Wang](https://github.com/orgs/jazzband/teams/members/discussions/18?from_comment=2#discussion-18-comment-2), we are adding a pre-commit CI configuration file.

### Actions performed by CI
- Strip unnecessary `# noqa`s
- isort
- Check for merge conflicts
- black
- Fix End of Files
- Fix requirements.txt
- Trim Trailing Whitespace
- File Contents Sorter
- Check for case conflicts
- Check JSON
- Check Xml
- Check Toml
- Check Xml
- Check Yaml
- Debug Statements (Python)
- Check for added large files
- Check for broken symlinks
- Debug Statements (Python)
- Detect AWS Credentials
- Detect Private Key
- pyupgrade
- flake8
- Check .rst files with the same linter used by pypi.org
